# Execinfo is not available on non-glibc systems (at least not on musl), so
# check the presence of the header before using the it for backtrace
# Little modified version of void linux patch, please refer here:
# https://github.com/void-linux/void-packages/blob/master/srcpkgs/blender/patches/0001-musl-fixes.patch
--- a/source/blender/blenlib/CMakeLists.txt
+++ b/source/blender/blenlib/CMakeLists.txt
@@ -1,6 +1,12 @@
 # SPDX-License-Identifier: GPL-2.0-or-later
 # Copyright 2006 Blender Foundation. All rights reserved.

+INCLUDE(CheckIncludeFiles)
+CHECK_INCLUDE_FILES("execinfo.h" HAVE_EXECINFO)
+IF (HAVE_EXECINFO)
+	add_compile_definitions(HAVE_EXECINFO_H)
+ENDIF(HAVE_EXECINFO)
+
 set(INC
   .
   # ../blenkernel  # don't add this back!
--- a/source/blender/blenlib/intern/system.c
+++ b/source/blender/blenlib/intern/system.c
@@ -21,7 +21,9 @@

 #  include "BLI_winstuff.h"
 #else
-#  include <execinfo.h>
+#  if defined(HAVE_EXECINFO_H)
+#    include <execinfo.h>
+#  endif
 #  include <unistd.h>
 #endif

@@ -63,7 +65,7 @@ void BLI_system_backtrace(FILE *fp)
 {
   /* ------------- */
   /* Linux / Apple */
-#  if defined(__linux__) || defined(__APPLE__)
+#  if defined(__linux__) && defined(HAVE_EXECINFO_H) || defined(__APPLE__)

 #    define SIZE 100
   void *buffer[SIZE];
